packages/babel-plugin-proposal-object-rest-spread/test/fixtures/regression/gh-7304/output.js

```diff
@@ -0,0 +1,15 @@
+"use strict";
+
+Object.defineProperty(exports, "__esModule", {
+  value: true
+});
+exports.default = void 0;
+
+class _default {
```

Comment:
This output seems to be buggy regarding what we discuss (cyclic deps and all) under babel-helper-split-export-declaration PR, right?

Reply:
I don't think that class declarations are hoisted? :thinking: 
```js
new A(); // ReferenceError
class A {}:
```

Anyway, that code is not related to this PR, it is generated either by transform-classes or transform-modules-commonjs.

Reply:
after experimenting with modules this output seems correct anyway